Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1325,42 +1325,75 @@ extension Tensor.IndexPath {
 //===------------------------------------------------------------------------------------------===//
 
 extension Tensor {
-  /// Returns `true` if the given axis is in the range `[-rank, rank)`.
+  /// Returns `true` iff `axis` can refer to an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange<T: BinaryInteger>(_ axis: T) -> Bool {
+  internal func isValid<T: BinaryInteger>(axis: T) -> Bool {
     let axis = Int(axis)
```

Comment:
I just noticed this.  You shouldn't need to convert the type of axis (and theoretically risk an out-of-range trap) if `rank` is an integer type.  Swift has heterogeneous numeric comparisons that actually work :-)

Reply:
Sorry, what do you mean here? I'm confused.